### PR TITLE
workflows

### PR DESCRIPTION
**Pull request by mathematicalmichael**

### CODE REVIEW

File: pants.toml [hunk 1]
```diff
@@ -18,15 +28,39 @@ verbosity=0
...
+args = ["--max-line-length 88"]
+
+[mypy]
+extra_type_stubs.add = [
+    "types-setuptools",
+    "types-dataclasses",
+    "types-requests",
+    "pandas-stubs==1.2.0.28",
+    "boto3-stubs==1.20.8"
+]
```
Comment (mathematicalmichael) [inline pants.toml]:
```suggestion
extra_type_stubs.add = [
    "types-setuptools",
    "types-dataclasses",
]
# for potential future use:
# "types-requests", "pandas-stubs", "boto3-stubs"
```

File: pants.toml [hunk 2]
```diff
@@ -18,15 +28,39 @@ verbosity=0
...
+lockfile = "<none>"
+config = "mypy.ini"
+
+# TODO: IS THIS USED?
+[test]
+use_coverage = false
+
+[python-repos]
```
Comment (mathematicalmichael) [inline pants.toml]:
```suggestion

# List of valid python package index locations:
[python-repos]
```

**Comment by sarodz** [comment]
Look into pages output

File: pants.toml [hunk 3]
```diff
@@ -18,15 +28,39 @@ verbosity=0
...
+#lockfile="3rdparty/python/pytest.lockfile"
+lockfile = "<none>"
+extra_requirements.add = [
+    "pytest-cov~=3.0.0",
+    "pytest-mock~=3.6.1",
+    "requests-mock~=1.9.3",
+]
+args = [
+#    "--cov", "base",
+#    "--cov-report", "term-missing",
```
Comment (mathematicalmichael) [inline pants.toml]:
```suggestion
```

Comment (mathematicalmichael) [reply pants.toml]: do codecov like this: https://www.pantsbuild.org/docs/python-test-goal#coverage

File: pants.toml [hunk 4]
```diff
@@ -18,15 +28,39 @@ verbosity=0
...
+    "types-dataclasses",
+]
+# for potential future use:
+# "types-requests", "pandas-stubs", "boto3-stubs"
+lockfile = "<none>"
+config = "mypy.ini"
+
+# TODO: IS THIS USED?
+[test]
+use_coverage = false
```
Comment (mathematicalmichael) [inline pants.toml]:
```suggestion
# report coverage on files encountered during tests
[test]
use_coverage = true
```

File: pants.toml [hunk 5]
```diff
@@ -18,15 +28,37 @@ verbosity=0
 
 [pytest]
 version = "pytest>=6.0"
-lockfile="3rdparty/python/pytest.lockfile"
+#lockfile="3rdparty/python/pytest.lockfile"
+lockfile = "<none>"
+extra_requirements.add = [
+    "pytest-cov~=3.0.0",
```
Comment (mathematicalmichael) [inline pants.toml]:
```suggestion
```
do we need this?

**Review by mathematicalmichael — COMMENTED** [review]
lgtm

**Review by sarodz — APPROVED** [review]

**Pull request merged**